NDTensors/src/abstractarray/fill.jl

```diff
@@ -1,19 +1,17 @@
-function generic_randn(arraytype::Type{<:AbstractArray}, dim::Integer=0)
+function generic_randn(
+  arraytype::Type{<:AbstractArray}, dim::Integer=0; rng=Random.default_rng()
+)
   arraytype_specified = set_unspecified_parameters(
     leaf_parenttype(arraytype), DefaultParameters()
   )
   data = similar(arraytype_specified, dim)
-  ElT = eltype(data)
-  for i in 1:length(data)
-    data[i] = randn(ElT)
-  end
-  return data
+  return randn!(rng, data)
 end
 
 function generic_zeros(arraytype::Type{<:AbstractArray}, dim::Integer=0)
   arraytype_specified = set_unspecified_parameters(
     leaf_parenttype(arraytype), DefaultParameters()
   )
   ElT = eltype(arraytype_specified)
-  return fill!(similar(arraytype_specified, dim), zero(ElT))
+  return arraytype_specified(zeros(ElT, dim))
```

Comment:
This doesn't seem great, since then it might create the data on CPU and then move it to GPU.

Can we use:
```julia
  data = similar(arraytype_specified, dim)
  return zeros!(data)
```
like we do for `generic_randn`?

Reply:
EDIT: I forgot there isn't a `zeros!` function but you can use `fill!`, which I see is what the previous version of this code is doing, was there an issue with that?

Reply:
I could imagine it might fail in that way for certain wrapped `CuArray`s, but we unwrap with `leaf_parenttype(arraytype)` so that shouldn't be a problem here.